### DISCUSSION TITLE
ESP32-S3-DevKitC-1-N8R8 and SPI

### DISCUSSION THREAD

**Discussion by carlberg74** [espressif/arduino-esp32#7153] [General]
Hi,

Trying to get a SPI to work on my new dev board, but I cant figure out what pins I should use for SPI on this board.

Have a MCP23S08 and used it with Arduino on other boards and are now trying to get it to work on this board.

I have read that SPI0 and SPI1 is used by flash etc.. and that SPI2 and SPI3 is available for other stuff and they are called HSPI (SPI2) and VSPI (SPI3)?

But looking on this layout, I dont really understand what pins to use..
https://docs.espressif.com/projects/esp-idf/en/latest/esp32s3/_images/ESP32-S3_DevKitC-1_pinlayout_v1.1.jpg
It has FSPICLK (11) and FSPICS0 (9) but no MISO or MOSI..  and is this SPI2 or SPI3?

Looked also at this.
https://github.com/espressif/esp-idf/blob/master/components/soc/esp32s3/include/soc/spi_pins.h

SPI2_CS   10
SPI2_MOSI 11
SPI2_CLK  12
SPI2_MISO 13
But that dont seem to work..

Any suggestions would be appreciated.

Cheers
-Anders

**Comment by carlberg74** [comment]
Just found this:
https://www.espressif.com/sites/default/files/documentation/esp32-s3_technical_reference_manual_en.pdf

FSPID = MISO
FSPIQ = MOSI 
And those are the pins I have been using...
Do I need to configure the dev board first using espressif tools to get SPI to work?

**Reply by sukhdarshan-sukhe** [reply]
![image](https://github.com/espressif/arduino-esp32/assets/85351114/772a4081-0620-4b5d-8daf-9c52d0c2ed8e)
As per this image FSPID = MOSI and FSPIQ = MISO.

**Comment by karoCB** [comment]
[Change resistor for SPI ?](https://www.lilygo.cc/products/t-display-s3-amoled#:~:text=T-Display-S3%20AMOLED%20is%20a%20display%20screen%20module%20with,is%20compatible%20with%20any%20microcontroller%20with%20SPI%20capability.)

and so ok ? 
SPI2_CS 10
SPI2_MOSI 13 FSPIQ = MOSI
SPI2_CLK 12
SPI2_MISO 11  FSPID = MISO
